model/task/task.go

```diff
@@ -454,9 +459,6 @@ func (t *Task) populateDependencyTaskCache(depCache map[string]Task) ([]Task, er
 		if err != nil {
 			return nil, errors.WithStack(err)
 		}
-		if len(newDeps) != len(depIdsToQueryFor) {
```

Comment:
In the case of non-existent dependencies, should we really include them? Also, should we log in this case, since presumably there's a bug? Or does removing the `fromNode` case in model/task_queue_service_dependency.go mean we never hit this?

Reply:
> In the case of non-existent dependencies, should we really include them?

I don't understand what you mean.

We will still hit this, and we could log it here too, but I figured since we call this function in preparation for calling `DependenciesMet` don't need to log it twice.